File: Facade/FacadeRestaurant/FacadeRestaurant/PizzaFacade.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace FacadeRestaurant
{
    public class PizzaFacade
    {
        private InventoryService _inventoryService;
        private KitchenService _kitchenService;
        private BillingService _billingService;
        private DeliveryService _deliveryService;

        public PizzaFacade()
        {
            _inventoryService = new InventoryService();
            _kitchenService = new KitchenService();
            _billingService = new BillingService();
            _deliveryService = new DeliveryService();
        }

        public void OrderPizza(string pizzaType, string address, decimal amount) 
        {
            Console.WriteLine("Starting order...");
            if (!_inventoryService.CheckIngredients(pizzaType)) 
            {
                Console.WriteLine("No ingredients available!");
                return;
            }

            _kitchenService.CookPizza(pizzaType);
            _billingService.ProcessPayment(pizzaType, amount);
            _deliveryService.DeliverPizza(address);

            Console.WriteLine("Order completed!");
        }
    }
}

File: Facade/JavaExample/FacadePizza/src/PizzaFacade.java
import java.math.BigDecimal;

public class PizzaFacade {
    private InventoryService inventoryService;
    private KitchenService kitchenService;
    private BillingService billingService;
    private DeliveryService deliveryService;

    public PizzaFacade() {
        this.inventoryService = new InventoryService();
        this.kitchenService = new KitchenService();
        this.billingService = new BillingService();
        this.deliveryService = new DeliveryService();
    }

    public void orderPizza(String pizzaType, String address, BigDecimal amount) {
        System.out.println("Starting order...");

        if (!inventoryService.checkIngredients(pizzaType)) {
            System.out.println("No ingredients available!");
            return;
        }

        kitchenService.cookPizza(pizzaType);
        billingService.processPayment(pizzaType, amount);
        deliveryService.deliverPizza(address);

        System.out.println("Order completed!");
    }
}
